src/components/ContactsList/ContactsTable.tsx/ContactsTable.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  LargeTd,
+  LargeTh,
+  MediumTd,
+  MediumTh,
+  NoDataInfo,
+  Table,
+  TableWrapper,
+} from '@/components/TransactionHistory/TransactionTable/TransactionTable.styles'
+import { SetPaymailButton } from '@/components/TransferForm/SetPaymailButton'
+import { FC, useMemo, useState } from 'react'
+import { StatusBadge } from './StatusBadge'
+import { VerifyModal } from '../_modals'
+import { SmallButton } from '@/components/Button'
+import { AcceptReject } from '../AcceptReject'
+import { useContacts } from '@/providers'
+import { ErrorBar } from '@/components/ErrorBar'
+import { useSortedContacts } from './useSortedContacts'
+import { JustAddedContactMsg } from './JustAddedContcatMsg'
+
+export const ContactsTable: FC = () => {
+  const { contacts, loading, error, refresh } = useContacts()
+  const [contactIdForVerification, setContactIdForVerification] = useState<string | null>(null)
+  const [justAddedContact, setJustAddedContact] = useState(false)
+
+  const contactForVerification = useMemo(() => {
+    return contacts?.find((contact) => contact.paymail === contactIdForVerification)
+  }, [contactIdForVerification, contacts])
+
+  const openVerificationWindow = (peerPaymail: string, justAdded = false) => {
+    setJustAddedContact(justAdded)
+    setContactIdForVerification(peerPaymail)
+  }
+
+  const sortedContacts = useSortedContacts(contacts)
+
+  if (error) {
+    return <ErrorBar errorMsg="Failed to load contacts" />
+  }
+
+  return (
+    <>
+      {loading && <Loader />}
+      <TableWrapper>
+        {!sortedContacts ? (
+          <NoDataInfo>No contacts yet</NoDataInfo>
+        ) : (
+          <Table>
+            <thead>
+              <tr>
+                <LargeTh>Paymail</LargeTh>
+                <MediumTh>Name</MediumTh>
+                <MediumTh>Status</MediumTh>
+                <MediumTh>Actions</MediumTh>
+              </tr>
+            </thead>
+            <tbody>
+              {sortedContacts.map((contact) => (
+                <tr key={contact.paymail} style={{ height: 50 }}>
+                  <LargeTd>{contact.paymail}</LargeTd>
+                  <MediumTd>{contact.name}</MediumTd>
+                  <MediumTd>
+                    <StatusBadge status={contact.status} />
+                  </MediumTd>
+                  <MediumTd>
+                    {contact.status !== ContactAwaitingAcceptance ? (
+                      <SmallButton
+                        variant="accept"
+                        onClick={() => {
+                          openVerificationWindow(contact.paymail)
+                        }}
+                      >
+                        Show code
+                      </SmallButton>
+                    ) : (
+                      <AcceptReject
+                        contact={contact}
+                        onAccept={() => {
+                          openVerificationWindow(contact.paymail, true)
+                          refresh()
+                        }}
```

Comment:
It's good to have own handler for this function.

`const handleAccept = (paymail: string) => {
openVerificationWindow(paymail, true)
                          refresh()
}`

Reply:
Generally I would agree. But in this case I could not do that, because there is also "accept" `onClick` a few lines above it. 
So in this case I see the inline function more readable than what you're suggesting.